src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -489,14 +489,23 @@ bool is_constant_and_all_values_equal_int(const Output<Node>& output, const int6
     return false;
 }
 
-bool is_on_constant_path(const ov::Output<ov::Node>& output) {
+bool is_on_constant_path(const ov::Output<ov::Node>& output,
+                         const std::unordered_set<std::type_index>& break_node_types) {
     auto status = true;
     std::deque<ov::Node*> nodes_to_calculate = {output.get_node()};
```

Comment:
minor (optional): it looks like, the function doesn't contain the typical
"is_used_node_check" before adding it to the deque

it would be nice if you help to fix this gap

Reply:
Done

Reply:
added the "visited" check